docs/README.md

```diff
@@ -753,16 +753,151 @@ public static async handle(event: StockMoved, register: Register): Promise<void>
+
+Entities are created on the fly, by _reducing_ the whole event stream. They aren't saved anywhere, although Booster
+does create automatic snapshots in order to make the reduction process efficient. You are the one that defines the
+reducer function.
+
+An entity is defined as a class with the `@Entity` decorator. Inside of it there is a static method with
+the `@Reduces` decorator that also specifies the event the reducer is subscribed to. In Booster an entity looks like this:
+
+```typescript
+@Entity
+export class EntityName {
+  public constructor(
+    readonly fieldA: SomeType,
+    readonly fieldB: SomeOtherType,
+    /* as many fields as needed */
+  ) {}
+
+  @Reduces(SomeEvent)
+  public static reduceSomeEvent(event: SomeEvent, previousState?: EntityName): EntityName {
+    /* Return a new entity based on the previous one */
+  }
+}
+```
+
+Each time an event is generated, all reducers that are listening to it are triggered. Note that event ordering is
+preserved per entity instance. This means that if two events arrive at the same time at a reducer, the first one to
+be picked will be the one that was generated first.
+
 #### Entities naming convention
 
+Entities model your data, so name them as closely to your domain as possible. Typical entity names are nouns that
+might appear when you think about your app. In an e-commerce application, some entities would be:
+
+* Cart
+* Product
+* UserProfile
+* Order
+* Address
+* PaymentMethod
+* Stock
+
+Entities live within the entities directory of the project source: `project-root/src/entities`.
+
+```text
+project-root
+├── src
+│   ├── commands
+│   ├── common
+│   ├── config
+│   ├── entities <------ They must be here
+│   ├── events
+│   ├── index.ts
+│   └── read-models
+```
+
 #### Creating entities
 
+The preferred way to create an entity is by using the generator, e.g.
+
+```text
+boost new:entity Product --fields displayName:string description:string price:Money
+```
+
+The generator will automatically create a file called `Product.ts` with a TypeScript class of the same name under the entities directory. You can still create the entity manually. Since the generator is not doing any magic, all you need is a class decorated as `@Command`. Anyway, we recommend you always to use the generator, because it handles the boilerplate code for you.
+
+Note:
+
+> Running the entity generator with an EntityName that already exists, it will override the content of the current one. In future releases, we will display a warning before overwriting anything. Meantime, if you missed a field, just add it to the class because in Booster, all the infrastructure and data structures are inferred from your code.
+
 #### The reducer function
 
+Booster generates the reducer function as a static method of the entity class. That function is called by the framework every time that an event of the
+specified type is emitted. **This function must be pure**, which means that no side effects should happen.
+
```

Comment:
@NickSeagull, could you please, define what a side effect is? Or rephrase this to say what to avoid explicitly?

Reply:
Done https://github.com/boostercloud/booster/pull/225/commits/4d592402bd59261283a79378ed0f969ee8710b79 thanks!